### PR TITLE
Feature/plcn 257 change hub spot id to crm

### PR DESCRIPTION
**Pull request by AntonEsbjergIT-Minds**

### CODE REVIEW

File: src/Pelican.Domain/Entities/Client.cs [hunk 1]
```diff
@@ -112,24 +115,27 @@ public virtual void FillOutClientContacts(IEnumerable<Contact>? contacts)
 
 
 	[GraphQLIgnore]
-	public virtual void UpdateClientContacts(ICollection<ClientContact>? currectHubSpotClientContacts)
+	public virtual void UpdateClientContacts(ICollection<ClientContact>? currentHubSpotClientContacts)
```
[the remark this answers is not in the document]
Comment (AntonEsbjergIT-Minds) [reply src/Pelican.Domain/Entities/Client.cs]: No longer contains HubSpot

File: src/Pelican.Domain/Entities/Client.cs [hunk 2]
```diff
@@ -112,24 +115,27 @@ public virtual void FillOutClientContacts(IEnumerable<Contact>? contacts)
...
+		if (currentHubSpotClientContacts is null)
 		{
 			return;
 		}
 
 		foreach (ClientContact clientContact in ClientContacts.Where(dc => dc.IsActive))
 		{
-			if (!currectHubSpotClientContacts.Any(currectHubSpotClientContact => currectHubSpotClientContact.HubSpotContactId == clientContact.HubSpotContactId))
+			if (!currentHubSpotClientContacts.Any(currentHubSpotClientContact => currentHubSpotClientContact.SourceContactId == clientContact.SourceContactId
+			&& currentHubSpotClientContact.Source == Sources.HubSpot))
```
[the remark this answers is not in the document]
Comment (AntonEsbjergIT-Minds) [reply src/Pelican.Domain/Entities/Client.cs]: No longer contains HubSpot in name

File: src/Pelican.Domain/Entities/Client.cs [hunk 3]
```diff
@@ -112,24 +115,27 @@ public virtual void FillOutClientContacts(IEnumerable<Contact>? contacts)
...
 			}
 		}
 
-		foreach (ClientContact clientContact in currectHubSpotClientContacts)
+		foreach (ClientContact clientContact in currentHubSpotClientContacts)
 		{
-			if (!ClientContacts.Any(dc => dc.HubSpotContactId == clientContact.HubSpotContactId && dc.IsActive))
+			if (!ClientContacts.Any(dc => dc.SourceContactId == clientContact.SourceContactId
+			&& dc.IsActive
+			&& dc.Source == Sources.HubSpot))
```
[the remark this answers is not in the document]
Comment (AntonEsbjergIT-Minds) [reply src/Pelican.Domain/Entities/Client.cs]: No longer contains HubSpot in name

File: src/Pelican.Domain/Entities/Contact.cs [hunk 4]
```diff
@@ -111,24 +113,27 @@ public virtual Contact UpdateProperty(string propertyName, string propertyValue)
 	}
 
 	[GraphQLIgnore]
-	public virtual void UpdateDealContacts(ICollection<DealContact>? currectHubSpotDealContacts)
+	public virtual void UpdateDealContacts(ICollection<DealContact>? currentHubSpotDealContacts)
```
[the remark this answers is not in the document]
Comment (AntonEsbjergIT-Minds) [reply src/Pelican.Domain/Entities/Contact.cs]: No longer contains HubSpot in name

File: src/Pelican.Domain/Entities/Contact.cs [hunk 5]
```diff
@@ -111,24 +113,27 @@ public virtual Contact UpdateProperty(string propertyName, string propertyValue)
...
+		if (currentHubSpotDealContacts is null)
 		{
 			return;
 		}
 
 		foreach (DealContact dealContact in DealContacts.Where(dc => dc.IsActive))
 		{
-			if (!currectHubSpotDealContacts.Any(currectHubSpotDealContact => currectHubSpotDealContact.HubSpotDealId == dealContact.HubSpotDealId))
+			if (!currentHubSpotDealContacts.Any(currentHubSpotDealContact => currentHubSpotDealContact.SourceDealId == dealContact.SourceDealId
+			&& currentHubSpotDealContact.Source == Sources.HubSpot))
```
[the remark this answers is not in the document]
Comment (AntonEsbjergIT-Minds) [reply src/Pelican.Domain/Entities/Contact.cs]: No longer contains HubSpot in name

File: src/Pelican.Domain/Entities/Contact.cs [hunk 6]
```diff
@@ -111,24 +113,27 @@ public virtual Contact UpdateProperty(string propertyName, string propertyValue)
...
 			}
 		}
 
-		foreach (DealContact dealContact in currectHubSpotDealContacts)
+		foreach (DealContact dealContact in currentHubSpotDealContacts)
 		{
-			if (!DealContacts.Any(dc => dc.HubSpotDealId == dealContact.HubSpotDealId && dc.IsActive))
+			if (!DealContacts.Any(dc => dc.SourceDealId == dealContact.SourceDealId
+			&& dc.IsActive
+			&& dc.Source == Sources.HubSpot))
```
[the remark this answers is not in the document]
Comment (AntonEsbjergIT-Minds) [reply src/Pelican.Domain/Entities/Contact.cs]: No longer contains HubSpot in name

File: src/Pelican.Domain/Entities/Contact.cs [hunk 7]
```diff
@@ -156,7 +161,7 @@ private void FillOutClient(IEnumerable<Client>? clients)
 			.Select(cc =>
 			{
 				Client? matchingClient = clients
-				.FirstOrDefault(client => client.HubSpotId == cc.HubSpotClientId);
+				.FirstOrDefault(client => client.SourceId == cc.SourceClientId && client.Source == Sources.HubSpot);
```
[the remark this answers is not in the document]
Comment (AntonEsbjergIT-Minds) [reply src/Pelican.Domain/Entities/Contact.cs]: No longer contains HubSpot in name

File: src/Pelican.Domain/Entities/Deal.cs [hunk 8]
```diff
@@ -149,7 +151,7 @@ private void FillOutDealContacts(List<Contact>? contacts)
 		foreach (DealContact dealContact in DealContacts)
 		{
 			Contact? matchingContact = contacts
-				.FirstOrDefault(contact => contact.HubSpotId == dealContact.HubSpotContactId);
+				.FirstOrDefault(contact => contact.SourceId == dealContact.SourceContactId && contact.Source == Sources.HubSpot);
```
[the remark this answers is not in the document]
Comment (AntonEsbjergIT-Minds) [reply src/Pelican.Domain/Entities/Deal.cs]: No longer contains HubSpot in name

File: src/Pelican.Domain/Entities/DealContact.cs [hunk 9]
```diff
@@ -13,17 +13,18 @@ public class DealContact : Entity, ITimeTracked
...
 
 
 	public Guid ContactId { get; set; }
 
-	public string HubSpotContactId { get; set; } = string.Empty;
+	public string SourceContactId { get; set; } = string.Empty;
 
 	public Contact Contact { get; set; }
 
+	public string Source { get; set; }
```
[the remark this answers is not in the document]
Comment (AntonEsbjergIT-Minds) [reply src/Pelican.Domain/Entities/DealContact.cs]: No it is not, has been removed.

File: src/Pelican.Infrastructure.HubSpot/Mapping/Deals/DealResponseToDeal.cs [hunk 10]
```diff
@@ -58,7 +62,7 @@ internal static Deal ToDeal(this DealResponse response)
 			.Where(association => association.Type == "deal_to_company")
 			.Select(association => new Client(Guid.NewGuid())
 			{
-				HubSpotId = association.Id,
+				SourceId = association.Id,
 				Deals = new List<Deal>() { result },
```
[the remark this answers is not in the document]
Comment (AntonEsbjergIT-Minds) [reply src/Pelican.Infrastructure.HubSpot/Mapping/Deals/DealResponseToDeal.cs:64]: Is now set here

File: src/Pelican.Infrastructure.HubSpot/Mapping/Clients/CompanyResponseToClient.cs [hunk 11]
```diff
@@ -29,7 +30,7 @@ internal static Client ToClient(this CompanyResponse response)
 			.Where(association => association.Type == "company_to_deal")
 			.Select(association => new Deal(Guid.NewGuid())
 			{
-				HubSpotId = association.Id,
+				SourceId = association.Id,
 				Client = result,
 				ClientId = result.Id,
 			})
```
[the remark this answers is not in the document]
Comment (AntonEsbjergIT-Minds) [reply src/Pelican.Infrastructure.HubSpot/Mapping/Clients/CompanyResponseToClient.cs:37]: Is now set here

**Review by Nicolai-Bak — APPROVED** [review]

**Pull request merged**